### DISCUSSION TITLE
How to make intellisense work for monaco editor with java language server

### DISCUSSION THREAD

**Discussion by chaitanya71998** [TypeFox/monaco-languageclient#450] [Q&A]
Hi,
I was working on JAVA project. I was using [@monaco-editor/react ](https://github.com/suren-atoyan/monaco-react) as the editor to show the file content in the web. I am giving project absolute path to `-data` value for the Eclipse JDT Language Server.  
In the editor, there is a prop for path, Here if i give the absolute file path of the file which will be in my local system, i am able to get intellisense working. But in ideal case, for every system this wont be case right. So for a project like system how to setup and give the path to the editor to make intellisense work.

**Comment by rick-chou** [comment]
@chaitanya71998  Hi , I have set `-data` but got error

![pic](https://user-images.githubusercontent.com/56244178/251360043-7dc7b368-2e6c-4897-aa40-47ea8fea2cc0.png)

**Comment by chaitanya71998** [comment]
@rick-chou check this repo https://github.com/chaitanya71998/monaco-language-testing-with-java

**Reply by rick-chou** [reply]
Thx!. But it seems to have the same problem
<img width="946" alt="Screenshot 2023-07-13 at 13 41 27" src="https://github.com/TypeFox/monaco-languageclient/assets/56244178/6c9f102d-01f5-4f69-8405-b7e6f0ced86e">

**Reply by rick-chou** [reply]
but yours can get code hints

**Reply by chaitanya71998** [reply]
Are you using java 11?

**Reply by rick-chou** [reply]
I'm using Java 16, the same version as JLS.

**Reply by dhruvk1989** [reply]
Did you guys find a solution? I;m stuck at the same place as @rick-chou...pls help me out @chaitanya71998

**Comment by dhruvk1989** [comment]
> Hi, I was working on JAVA project. I was using [@monaco-editor/react ](https://github.com/suren-atoyan/monaco-react) as the editor to show the file content in the web. I am giving project absolute path to `-data` value for the Eclipse JDT Language Server. In the editor, there is a prop for path, Here if i give the absolute file path of the file which will be in my local system, i am able to get intellisense working. But in ideal case, for every system this wont be case right. So for a project like system how to setup and give the path to the editor to make intellisense work.

Hi. How did you integrate the eclipse lsp with Monaco? Can you share steps or something? I do the initialize but can't understand what to do next. I haven't even reached the Monaco part. Still struggling with figuring out how to make it work in terminal itself.

**Reply by chaitanya71998** [reply]
@dhruvk1989 check this :https://github.com/chaitanya71998/monaco-language-testing-with-java

**Reply by dhruvk1989** [reply]
Hi @chaitanya71998 tried that...I've left comments on another thread as well after trying that library. It's not recognising the file as part of a project and highlighting only syntax errors.

You have a thread open for that. I've replied there.

**Reply by kaisalmen** [reply]
We have a full example available: https://github.com/TypeFox/monaco-languageclient#java-language-client-and-language-server-example-location
